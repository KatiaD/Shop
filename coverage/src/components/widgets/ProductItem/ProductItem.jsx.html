<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/components/widgets/ProductItem/ProductItem.jsx</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../../prettify.css" />
    <link rel="stylesheet" href="../../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../../index.html">All files</a> / <a href="index.html">src/components/widgets/ProductItem</a> ProductItem.jsx
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">80% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>4/5</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/2</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">0% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>0/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">80% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>4/5</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-yes">3x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'components/controls';
import {
  ProductContainer, ProductName, ProductPriceBlock, ProductPrice,
} from './ProductItem.styled';
&nbsp;
const displayName = 'ProductItem';
&nbsp;
const propTypes = {
  name: PropTypes.string.isRequired,
  handleAddToCart: PropTypes.func.isRequired,
};
&nbsp;
function <span class="fstat-no" title="function not covered" >ProductItem(</span>{
  name, image, handleAddToCart, price,
}) {
<span class="cstat-no" title="statement not covered" >  return (</span>
    name &amp;&amp; (
      &lt;ProductContainer data-name={name}&gt;
        &lt;img src={image} alt="" /&gt;
        &lt;ProductName&gt;
          {name}
        &lt;/ProductName&gt;
        &lt;ProductPriceBlock&gt;
          &lt;ProductPrice&gt;
            {price}
            {' '}
          $
          &lt;/ProductPrice&gt;
          &lt;Button accent onClick={handleAddToCart}&gt;
            Buy
          &lt;/Button&gt;
        &lt;/ProductPriceBlock&gt;
      &lt;/ProductContainer&gt;
    )
  );
}
&nbsp;
ProductItem.displayName = displayName;
ProductItem.propTypes = propTypes;
&nbsp;
export default ProductItem;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Nov 26 2018 16:35:58 GMT+0200 (EET)
</div>
</div>
<script src="../../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../../sorter.js"></script>
</body>
</html>
