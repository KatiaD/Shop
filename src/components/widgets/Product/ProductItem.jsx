import React from 'react';
import PropTypes from 'prop-types';

import { Button } from 'components/controls';

const displayName = 'ProductItem';

const propTypes = {
  name: PropTypes.string.isRequired,
  price: PropTypes.string.isRequired,
  handleAddToCart: PropTypes.func.isRequired,
};


function Product({
  id,
  name,
  price,
  onAddToCart,
}) {
  return (
    name && (
      <div data-name={name}>
        {id}
        {name}
        {price}
        <Button onClick={onAddToCart} value="addToCart">
          Bye
        </Button>
      </div>
    )
  );
}

Product.displayName = displayName;
Product.propTypes = propTypes;

export default Product;
