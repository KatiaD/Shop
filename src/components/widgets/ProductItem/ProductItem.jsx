import React from 'react';
import PropTypes from 'prop-types';
import { Button } from 'components/controls';
import {
  ProductContainer, ProductName, ProductPriceBlock, ProductPrice,
} from './ProductItem.styled';

const displayName = 'ProductItem';

const propTypes = {
  name: PropTypes.string.isRequired,
  handleAddToCart: PropTypes.func.isRequired,
};

function ProductItem({
  name, image, handleAddToCart, price,
}) {
  return (
    name && (
      <ProductContainer data-name={name}>
        <img src={image} alt="" />
        <ProductName>
          {name}
        </ProductName>
        <ProductPriceBlock>
          <ProductPrice>
            {price}
            {' '}
          $
          </ProductPrice>
          <Button accent onClick={handleAddToCart}>
            Buy
          </Button>
        </ProductPriceBlock>
      </ProductContainer>
    )
  );
}

ProductItem.displayName = displayName;
ProductItem.propTypes = propTypes;

export default ProductItem;
