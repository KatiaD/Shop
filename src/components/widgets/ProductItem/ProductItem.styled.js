import styled from "styled-components";

export const ProductWrapper = styled.div`
  display: flex;
  justify-content: space-around;
`;

export const ProductItemContainer = styled.div`
  img {
    height: 200px;
    width: 200px;
    border: 1px solid lightgray;
    border-radius: 10px;
    margin: 10px;
  }
`;
