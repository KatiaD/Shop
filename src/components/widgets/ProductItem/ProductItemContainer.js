import {
  compose, setDisplayName, withHandlers, setPropTypes,
} from 'recompose';
import PropTypes from 'prop-types';
import ProductItem from './ProductItem';

export const enhance = compose(
  setDisplayName('ProductItemContainer'),
  setPropTypes({
    id: PropTypes.number.isRequired,
    AddToCart: PropTypes.func.isRequired,
  }),
  withHandlers({
    handleAddToCart: ({ id, AddToCart }) => () => AddToCart(id),
  }),
);

export default enhance(ProductItem);
