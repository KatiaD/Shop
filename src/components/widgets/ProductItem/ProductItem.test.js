import React from 'react';
import { shallow } from 'enzyme';
import ProductItem from './ProductItem';

const testProps = {
  handleAddToCart: jest.fn(),
  name: 'GUNMETAL SANDSTONE',
  image: '',
  price: 140,
};
describe('Given the ProductItem component', () => {
  describe('when the component is rendered', () => {
    const component = shallow(<ProductItem {...testProps} />);
    it('should render the component with provided props', () => {
      expect(component).toMatchSnapshot();
    });
  });
});
