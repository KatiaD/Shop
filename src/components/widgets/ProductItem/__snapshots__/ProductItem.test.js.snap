// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given the ProductItem component when the component is rendered should render the component with provided props 1`] = `
<ProductItemstyled__ProductContainer
  data-name="GUNMETAL SANDSTONE"
>
  <img
    alt=""
    src=""
  />
  <ProductItemstyled__ProductName>
    GUNMETAL SANDSTONE
  </ProductItemstyled__ProductName>
  <ProductItemstyled__ProductPriceBlock>
    <ProductItemstyled__ProductPrice>
      140
       
      $
    </ProductItemstyled__ProductPrice>
    <Button
      accent={true}
      onClick={[MockFunction]}
    >
      Buy
    </Button>
  </ProductItemstyled__ProductPriceBlock>
</ProductItemstyled__ProductContainer>
`;
